### ISSUE TITLE
The training problem

### ISSUE THREAD

**Issue by f1yfisher** [hpcaitech/Open-Sora#542]
I found that the number of output channels of STDiT-v3 is 8, but in the training phase, only the former 4 channels are computed in  the loss function.

**Comment by JThh** [comment]
Thanks. Would you please help me pinpoint where you identified this issue? I saw [this](https://github.com/hpcaitech/Open-Sora/blob/278cd759e58aff49c4349e102584e7d1c15045bf/opensora/schedulers/iddpm/gaussian_diffusion.py#L738) but failed to locate where channel selection happened.

**Comment by f1yfisher** [comment]
The training of open sora v1.2 used the rectified flow scheduler in [this](https://github.com/hpcaitech/Open-Sora/blob/278cd759e58aff49c4349e102584e7d1c15045bf/opensora/schedulers/rf/rectified_flow.py).

**Comment by narrowsnap** [comment]
In the DiT training code, the purpose of setting sigma is to learn the mean and variance of the noise, and then calculate the KL loss with the gold standard. The code sets pred_sigma to True by default, but directly uses the mean (the first 4 channels) as the predicted noise.

**Comment by f1yfisher** [comment]
The model released is also 8 channels (including mean and variance). Why set the model to predict mean and variance, but do not calculate the loss of variance.

**Comment by narrowsnap** [comment]
> The model released is also 8 channels (including mean and variance). Why set the model to predict mean and variance, but do not calculate the loss of variance.

I also want to know. @JThh Is there any reason to do this?

**Comment by henbucuoshanghai** [comment]
> In the DiT training code, the purpose of setting sigma is to learn the mean and variance of the noise, and then calculate the KL loss with the gold standard. The code sets pred_sigma to True by default, but directly uses the mean (the first 4 channels) as the predicted noise.

 so why?the output dimensions is? input is ?

**Comment by xesdiny** [comment]
I felt the same doubt.

**Comment by narrowsnap** [comment]
> I felt the same doubt. why Just use out_channel=8 ,uses the mean (the first 4 channels) ,dorpout bias? Maybe,Update latent_z by grad without bias
> 
> ```
>       dt = timesteps[i] - timesteps[i + 1] if i < len(timesteps) - 1 else timesteps[i]
>       dt = dt / self.num_timesteps
>       z = z + v_pred * dt[:, None, None, None, None]
> ```

This issue come from pixart-alpha which align with the original DiT. (https://github.com/PixArt-alpha/PixArt-sigma/issues/81#issuecomment-2100610843)

**Comment by henbucuoshanghai** [comment]
i see
